docs/configuration/config-file-reference.md

```diff
@@ -2438,6 +2438,37 @@ grpc_client_config:
   # Skip validating server certificate.
   # CLI flag: -querier.frontend-client.tls-insecure-skip-verify
   [tls_insecure_skip_verify: <boolean> | default = false]
+
+memory_limiter:
+  # Enable memory-limiter.
+  # CLI flag: -querier.memory-limiter.enabled
+  [enabled: <boolean> | default = false]
+
+  # Specifies whether to fail when the heap is above the limit. If fail fast is
+  # false, the requests will wait for the heap to go down or until the
+  # max-wait-duration.
+  # CLI flag: -querier.memory-limiter.fail-fast
+  [fail_fast: <boolean> | default = false]
+
+  # Specifies the limit above which system is considered to in red state. In red
+  # state, all except one request will be throttled.
```

Comment:
I should change it to "all except one request will be slowed down". The return code would be on the following:
- 500 if fail fast is true.
- 200 if the request can finish before timing out.
- 504 if the requests times out before it can finish.